pkg/prometheus/rules.go

```diff
@@ -203,6 +207,39 @@ func (c *Operator) selectRules(p *monitoringv1.Prometheus, namespaces []string)
 	return rules, nil
 }
 
+func generateContent(promRule monitoringv1.PrometheusRuleSpec, enforcedNsLabel, ns string) (string, error) {
+	if enforcedNsLabel != "" {
+		for gi, group := range promRule.Groups {
+			for ri, r := range group.Rules {
+				if len(promRule.Groups[gi].Rules[ri].Labels) == 0 {
+					promRule.Groups[gi].Rules[ri].Labels = map[string]string{}
+				}
+				promRule.Groups[gi].Rules[ri].Labels[enforcedNsLabel] = ns
+
+				expr := r.Expr.String()
+				parsedExpr, err := promql.ParseExpr(expr)
+				if err != nil {
+					return "", err
```

Comment:
admittedly, I haven't checked the rest of the code, but we add some `errors.Wrap`ish context when returning the error? It could ease debugging in log files.